### PR TITLE
bim.select_aggregate(select_parts=True) - selects all the host aggreg…

### PR DESCRIPTION
**Pull request by theoryshaw**
…ates and all the parts and subaggregates within these host aggregates

bim.select_linked_aggregates(select_parts=False) - selects all the host aggregates that are linked bim.select_linked_aggregates(select_parts=True) - selects all the host aggregates that are linked and all the parts and subaggregates within these host aggregates

### CODE REVIEW

**Comment by theoryshaw** [comment]
This PR applies the same functionality proposed in [this](https://github.com/IfcOpenShell/IfcOpenShell/pull/4437) old PR (now closed), and adopts @Gorgious56 UI logic he applied on [this](https://github.com/IfcOpenShell/IfcOpenShell/commit/4dda7e84a0fae6e5831ea06cf2253277dda628fb) commit. 

...

@brunoperdigao 
I tried to use the code you suggested [here](https://github.com/IfcOpenShell/IfcOpenShell/pull/4437#discussion_r1529602674), but it did not select subaggregates and parts within those subaggregates--only the parts in the immediate, top level aggregate.

**Comment by theoryshaw** [comment]
@Gorgious56 In this PR i'm proposing to extend `bim.select_aggregate(select_parts=True)` to select everything within the host aggregate -- that is all subaggregates and all their associated parts.

**Comment by theoryshaw** [comment]
A quick video to show functionality...

https://github.com/IfcOpenShell/IfcOpenShell/assets/507113/db9b18f3-6260-4e04-9ff8-ec382b7cd8b4

**Comment by theoryshaw** [comment]
a quick video to show functionality. 

https://github.com/IfcOpenShell/IfcOpenShell/assets/507113/97186a14-7618-4389-a4a5-9686790962be

File: src/blenderbim/blenderbim/bim/module/aggregate/operator.py [hunk 1]
```diff
@@ -190,12 +190,17 @@ class BIM_OT_select_parts(bpy.types.Operator):
...
-        selectable_parts_objs = set(context.selectable_objects).intersection(parts_objs)
-        for selectable_part_obj in selectable_parts_objs:
-            selectable_part_obj.select_set(True)
+        # obj = bpy.data.objects.get(self.obj) or context.active_object
+
+        for obj in context.selected_objects:
+            parts = ifcopenshell.util.element.get_parts(tool.Ifc.get_entity(obj))
+            if parts:
+                p_objs = set(tool.Ifc.get_object(part) for part in parts)
+                selectable_p_objs = set(context.selectable_objects).intersection(p_objs)
```
Comment (brunoperdigao) [inline src/blenderbim/blenderbim/bim/module/aggregate/operator.py:199]: My only comment would be to keep the name of the variables for clarity. `parts_objs` and `selectable_parts_objs`.

Comment (brunoperdigao) [inline src/blenderbim/blenderbim/bim/module/aggregate/operator.py:199]: @theoryshaw My only comment would be to keep the name of the variables for clarity. `parts_objs` and `selectable_parts_objs`.

Comment (theoryshaw) [reply src/blenderbim/blenderbim/bim/module/aggregate/operator.py]: Thanks!

**Pull request merged**